### PR TITLE
dependency: remove clippy

### PR DESCRIPTION
**Pull request by BusyJay**
Because version '*' is rejected by crates.io, so clippy needs to be opt out from dependency. The pr uses the command mode of clippy.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,11 +18,6 @@ log = "0.3"
 protobuf = "1.2"
 quick-error = "0.2"
 rand = "0.4"
-clippy = {version = "*", optional = true}
```
Comment (siddontang) [inline Cargo.toml]: you can use `clippy = {version = "^0", optional = true}` instead to pass the crate like rust prometheus does.

Comment (BusyJay) [reply Cargo.toml]: I prefer the current way. So `allow_failure` is unnecessary anymore.

Comment (siddontang) [reply Cargo.toml]: how can we use clippy in local machine now?

Comment (BusyJay) [reply Cargo.toml]:
```
- cargo +nightly-2018-01-12 install -f clippy
- cargo +nightly-2018-01-12 clippy
```

Comment (BusyJay) [reply Cargo.toml]: Or you can just do nothing and let CI lints your code.

Comment (siddontang) [reply Cargo.toml]: Got it

File: .travis.yml [hunk 2]
```diff
@@ -3,21 +3,21 @@ sudo: false
...
 matrix:
-  allow_failures:
-    # clippy (behind the 'dev' feature) only works on the nightly toolchain.
-    # This build is considerably more thorough.
-    - rust: nightly
   include:
   - rust: stable
-  - rust: nightly
-    env: FEATURES="--all-features"
+  - rust: nightly-2018-01-01
```
Comment (siddontang) [inline .travis.yml]: why not using our nightly version?

**Comment by siddontang** [comment]
LGTM

**Pull request merged**